src/drawing-tool/components/drawing-tool.tsx

```diff
@@ -56,6 +59,15 @@ export const DrawingTool: React.FC<IProps> = ({ authoredState, interactiveState,
     let backgroundImgSrc: string | undefined;
     if (authoredState.backgroundSource === "url") {
       backgroundImgSrc = authoredState.backgroundImageUrl;
+      try {
+        const url = new URL(backgroundImgSrc || "");
+        if (!url.host.match(/concord\.org/)) {
+          // Use LARA image proxy to avoid tainting canvas when external image URL is used.
```

Comment:
I think it is also required that the image needs to have CORS headers. Not every site at CC will have those.
I think the approach above is still OK, but I think there should be a comment here about the CORS headers being needed. 

If there are no CORS headers, what happens? Will the drawing not save, or will there be an error when the background is first loaded?

Reply:
> I think it is also required that the image needs to have CORS headers. Not every site at CC will have those.

Right, I was assuming that if we test against concord.org, we'll allow sites like models-resources.concord.org and similar. But this is not 100% correct of course. Just a compromise to avoid using proxy all the time.

> If there are no CORS headers, what happens? Will the drawing not save, or will there be an error when the background is first loaded?

This is handled by the Drawing Tool library. First, it'll try to request image with crossOrigin=anonymous. This request will fail if the server doesn't have CORS headers. In this case, it'll print a warining in the console and request an image again without crossOrigin headers. The background will be set and visible for users, but saving will fail.

Drawing Tool could fail even earlier, after the first request. But author / user would still need to open the image question dialog to see that something is broken. And it seems it's not really a concern of Drawing Tool app, as it doesn't really know if anyone would like to use `.toDataUrl()` (although I think we're the only users of this lib, so we could adjust it a bit).

We could also add some more advanced validation to authoring interface. I think that react-json-schema-form allows async validation. In this case we could make a request even in the authoring UI.